### PR TITLE
Add notification and folder structure

### PR DESCRIPTION
**Pull request by Nuwa-Hub**

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
-  templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  selector: "app-root",
+  templateUrl: "./app.component.html",
+  styleUrls: ["./app.component.scss"],
 })
 export class AppComponent {
-  title = 'kendo-angular-app';
+  title = "kendo-angular-app";
+
```
Comment (ChamaraWee) [inline src/app/app.component.ts]: spaces??

File: src/app/models/user-interface.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface UserInterface {
+  username: string;
+
```
Comment (ChamaraWee) [inline src/app/models/user-interface.ts]: extra spaces?

File: src/app/sign-in-module/containers/sign-in/sign-in.component.scss [hunk 3]
```diff
@@ -0,0 +1,188 @@
+$color_1: #2daabd;
+$color_2: white;
+$font-family_1: "Raleway", sans-serif;
+$font-family_2: "Raleway Thin", sans-serif;
+$font-family_3: "Raleway SemiBold", sans-serif;
+
+.example-wrap {
```
Comment (ChamaraWee) [inline src/app/sign-in-module/containers/sign-in/sign-in.component.scss]: use meaning full names. For classes

File: src/app/sign-in-module/containers/sign-in/sign-in.component.scss [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  border-style: solid;
+  border-color: black !important;
+  border-radius: 5px;
+  margin: 5px;
+  padding: 5px;
+}
+.login-button-action {
+  margin: 5px 40px;
+  padding: 10px;
+
```
Comment (ChamaraWee) [inline src/app/sign-in-module/containers/sign-in/sign-in.component.scss]: extra spaces?

File: src/app/sign-in-module/containers/sign-in/sign-in.component.scss [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  cursor: pointer;
+  color: $color_2;
+  font-family: $font-family_3;
+  height: 42.3px;
+  transition: 0.25s;
+  width: 153px;
+  &:hover {
+    box-shadow: 0px 1px 18px #249bc6;
+  }
+}
```
Comment (ChamaraWee) [inline src/app/sign-in-module/containers/sign-in/sign-in.component.scss:180]: make whole file indentation consistence. in some space there is space between classes. Some places don't have any space. Your code reflects you.

File: src/app/table-module/containers/home-page/home-page.component.ts [hunk 6]
```diff
@@ -113,15 +147,17 @@ export class HomePageComponent {
...
       this.store.dispatch(personActions.addPersonstart({ personData }));
     } else {
       this.store.dispatch(personActions.updatePersonstart({ personData }));
     }
     sender.closeRow(rowIndex);
   }
   public removeHandler(args: RemoveEvent): void {
-    this.store.dispatch(//todo implement the backend api
+    // remove the current dataItem from the current data source, and close the row
+    //this.editService.remove(args.dataItem.PersonID);
```
Comment (ChamaraWee) [inline src/app/table-module/containers/home-page/home-page.component.ts]: remove dead code

File: src/app/table-module/containers/home-page/home-page.component.ts [hunk 7]
```diff
@@ -134,10 +170,20 @@ export class HomePageComponent {
   public onStateChange(state: State): void {
     this.gridState = state;
     this.store.dispatch(personActions.getPersonstart());
+    // this.editService.read();
```
Comment (ChamaraWee) [inline src/app/table-module/containers/home-page/home-page.component.ts]: ??

File: src/app/table-module/table-module.module.ts [hunk 8]
```diff
@@ -1,27 +1,42 @@
...
+    DropDownListModule,
+    DateInputsModule,
+    ReactiveFormsModule,
+    FormsModule,
+    PopupModule,
+    InputsModule,
+    BrowserAnimationsModule,
+    NotificationModule,
+  ],
+
```
Comment (ChamaraWee) [inline src/app/table-module/table-module.module.ts]: spaces?

File: src/app/utils/derectives/clickoutside.directive.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+})
+export class ClickoutsideDirective {
+
+ constructor(private elementRef: ElementRef) { }
+
+ @Output() public clickOutside = new EventEmitter<MouseEvent>();
+
+    @HostListener('document:click', ['$event', '$event.target'])
+    public onClick(event: MouseEvent, targetElement: HTMLElement): void {
+        if (!targetElement) {
```
Comment (ChamaraWee) [inline src/app/utils/derectives/clickoutside.directive.ts]: `if (!targetElement) return`

File: src/app/utils/derectives/clickoutside.directive.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+            return;
+        }
+        const clickedInside = this.elementRef.nativeElement.contains(targetElement);
+        if (!clickedInside) {
```
Comment (ChamaraWee) [inline src/app/utils/derectives/clickoutside.directive.ts]: same goes here

**Review by ChamaraWee — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts (hunk 1 above)
Comment (Nuwa-Hub) [reply src/app/app.component.ts]: done

File: src/app/models/user-interface.ts (hunk 2 above)
Comment (Nuwa-Hub) [reply src/app/models/user-interface.ts]: done

File: src/app/sign-in-module/containers/sign-in/sign-in.component.scss (hunk 3 above)
Comment (Nuwa-Hub) [reply src/app/sign-in-module/containers/sign-in/sign-in.component.scss]: done

File: src/app/sign-in-module/containers/sign-in/sign-in.component.scss (hunk 4 above)
Comment (Nuwa-Hub) [reply src/app/sign-in-module/containers/sign-in/sign-in.component.scss]: done

File: src/app/sign-in-module/containers/sign-in/sign-in.component.scss (hunk 5 above)
Comment (Nuwa-Hub) [reply src/app/sign-in-module/containers/sign-in/sign-in.component.scss:180]: done

File: src/app/table-module/containers/home-page/home-page.component.ts (hunk 6 above)
Comment (Nuwa-Hub) [reply src/app/table-module/containers/home-page/home-page.component.ts]: done

File: src/app/table-module/containers/home-page/home-page.component.ts (hunk 7 above)
Comment (Nuwa-Hub) [reply src/app/table-module/containers/home-page/home-page.component.ts]: done

File: src/app/table-module/table-module.module.ts (hunk 8 above)
Comment (Nuwa-Hub) [reply src/app/table-module/table-module.module.ts]: done

File: src/app/utils/derectives/clickoutside.directive.ts (hunk 9 above)
Comment (Nuwa-Hub) [reply src/app/utils/derectives/clickoutside.directive.ts]: done

**Pull request merged**